addons/linecompass/functions/fnc_cacheLoop.sqf

```diff
@@ -0,0 +1,104 @@
+#include "..\script_component.hpp"
+
+[{ call FUNC(cacheLoop); }, [], 0.5] call CBA_fnc_waitAndExecute;
+
+private _player = call CBA_fnc_currentUnit;
+
+private _canShow = [_player] call EFUNC(main,canHudBeShown);
+
+if (!_canShow || !GVAR(Enabled)) exitWith {
+    if (GVAR(CompassShown)) then {
+        call FUNC(HideCompass);
+    };
+};
+
+if (_canShow && GVAR(Enabled)) then {
+    if (!GVAR(CompassShown)) then {
+        call FUNC(ShowCompass);
+    };
+};
+
+if (GVAR(CompassAvailableShown) && {floor(time % 1) == 0}) then {
+    if (!GVAR(CompassShown) && { call EFUNC(main,getCompass) isNotEqualTo "" }) then {
```

Comment:
`EFUNC(main,getCompass)` requires a unit to be passed. Right now it will return nil, breaking the CompassAvailableShown setting to always show.

Reply:
Another thing is, once this is fixed, the line compass will just start blinking